webknossos/webknossos/geometry/bounding_box.py

```diff
@@ -293,6 +295,20 @@ def align_with_mag(self, mag: Mag, ceil: bool = False) -> "BoundingBox":
             bottomright = align(self.bottomright, np.floor)
         return BoundingBox(topleft, bottomright - topleft)
 
+    def align_with_mag2(self, mag: Mag, ceil: bool = False) -> "BoundingBox":
+        # TODO: consider alternative implementation using ints only  pylint: disable=fixme
+        mag_vec = mag.to_vec3_int()
+        roundup = self.topleft if ceil else self.bottomright
+        rounddown = self.bottomright if ceil else self.topleft
+        margin_to_roundup = roundup % mag_vec
+        aligned_roundup = roundup - margin_to_roundup
+        margin_to_rounddown = (mag_vec - (rounddown % mag_vec)) % mag_vec
+        aligned_rounddown = rounddown + margin_to_rounddown
+        if ceil:
+            return BoundingBox(aligned_roundup, aligned_rounddown - aligned_roundup)
+        else:
+            return BoundingBox(aligned_rounddown, aligned_roundup - aligned_rounddown)
```

Comment:
I took the alignment implementation from the old `View._handle_compressed_write` implementation and compared it to the numpy version, it's quite a bit faster. Not sure if it's readable enough, though, I think the initial version is easier to grasp.
```suggestion
    def align_with_mag2(self, mag: Mag, ceil: bool = False) -> "BoundingBox":
        # TODO: consider alternative implementation using ints only  pylint: disable=fixme
        mag_vec = mag.to_vec3_int()
        roundup = self.topleft if ceil else self.bottomright
        rounddown = self.bottomright if ceil else self.topleft
        margin_to_roundup = roundup % mag_vec
        aligned_roundup = roundup - margin_to_roundup
        margin_to_rounddown = (mag_vec - (rounddown % mag_vec)) % mag_vec
        aligned_rounddown = rounddown + margin_to_rounddown
        if ceil:
            return BoundingBox(aligned_roundup, aligned_rounddown - aligned_roundup)
        else:
            return BoundingBox(aligned_rounddown, aligned_roundup - aligned_rounddown)
```
```suggestion
    def align_with_mag2(self, mag: Mag, ceil: bool = False) -> "BoundingBox":
        # TODO: consider alternative implementation using ints only  pylint: disable=fixme
        mag_vec = mag.to_vec3_int()
        roundup = self.topleft if ceil else self.bottomright
        rounddown = self.bottomright if ceil else self.topleft
        margin_to_roundup = roundup % mag_vec
        aligned_roundup = roundup - margin_to_roundup
        margin_to_rounddown = (mag_vec - (rounddown % mag_vec)) % mag_vec
        aligned_rounddown = rounddown + margin_to_rounddown
        if ceil:
            return BoundingBox(aligned_roundup, aligned_rounddown - aligned_roundup)
        else:
            return BoundingBox(aligned_rounddown, aligned_roundup - aligned_rounddown)
```
```
>>> timeit("bb.align_with_mag(wk.Mag(2), ceil=True)", globals=globals(), number=100000)
3.7116026320145465
>>> timeit("bb.align_with_mag2(wk.Mag(2), ceil=True)", globals=globals(), number=100000)
1.4745627499651164
```

Reply:
I'll rename the old version to `align_with_mag_slow` and add some tests that assert both methods do the same.